packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -252,27 +252,32 @@ async function createNewDatasetFromFile(
 	return newDataset;
 }
 
+// If isTemporary = true this will not add the asset to the project.
+// If isTemporary is not provided or false this asset will be added to the project.
 async function createDatasetFromSimulationResult(
 	projectId: string,
 	simulationId: string,
-	datasetName: string | null
-): Promise<boolean> {
+	datasetName: string | null,
+	isTemporary?: boolean
+): Promise<Dataset | null> {
+	if (isTemporary === undefined) isTemporary = false;
 	try {
-		const response: AxiosResponse<Response> = await API.post(
-			`/simulations/${simulationId}/add-result-as-dataset-to-project/${projectId}?dataset-name=${datasetName}`
+		const response: AxiosResponse<Dataset> = await API.post(
+			`/simulations/${simulationId}/create-result-as-dataset/${projectId}?dataset-name=${datasetName}&is-temporary=${isTemporary}`
```

Comment:
We have an issue #4228 to remove the `isTemporary`.
So I'm not sure we need to do this, here.

Reply:
It was my understanding that when we run a forecast with a node we should create a _temporary_ dataset asset.
If the user then wants to save this asset into their project they would use this menu:
<img width="1077" alt="image" src="https://github.com/user-attachments/assets/2775798c-112d-4131-9a3e-5683f58535c8">

Which will create a copy of the dataset asset, marked as not temporary, and add this new asset to the project.

Do you have a quick minute to chat about this?

Reply:
Thanks Yohann!